src/app/clinic/clinic_facade.hpp

```diff
@@ -15,21 +16,25 @@ class Clinic
     inline static std::vector<std::shared_ptr<Visit>> visits_;
     inline static std::vector<std::shared_ptr<Receptionist>> receptionists_;
     inline static std::vector<std::shared_ptr<Room>> rooms_;
+    inline static std::shared_ptr<Calendar> calendar_;
```

Comment:
**idea**
As we spotted some issues with the static containers I suggest that in the future we will create an object Clinic which will take care of lifetime of other objects.
In such term, once Clinic will be created in main, we will have only one calendar initialized and then used by dependency injection. 
And then, if there will be single Calendar&, then we can remove checking if it's the only Calendar. There would not be a possibility to create another :D

Reply:
Yes, that's one way to do it, but what if we want to expand our business and make a new branch? We can consider adding support for multiple calendars with distinction to which branch it is from.